google/cloud/internal/backoff_policy.cc

```diff
@@ -40,8 +41,11 @@ std::chrono::milliseconds ExponentialBackoffPolicy::OnCompletion() {
   }
 
   if (current_delay_end_ >= maximum_delay_) {
-    current_delay_start_ = std::max(maximum_delay_ / scaling_, initial_delay_);
-    current_delay_end_ = maximum_delay_;
+    if (scaling_lower_bound_ != 1) {
+      current_delay_start_ =
+          (std::max)(minimum_delay_, current_delay_end_ / scaling_lower_bound_);
+    }
```

Comment:
> consider: ExponentialBackoffPolicy(ms(1), ms(10), ms(10), 2.0, 2.0)
>
> The range will be:
> [1, 10], [1, 10], [1, 10], etc.

argh, sorry. The range would be:
[5, 10], [5, 10], [5, 10], etc.

> `scaling_lower_bound == 1 addresses the minjitter case.`

Ok, but if `scaling_lower_bound == 1.0000000000000000000001`, the behavior will be very different. That's weird.

> Added the test case you mentioned

The test is passing because the actual ranges are subsets of the expected/desired ranges.

Consider:

```cc
TEST(ExponentialBackoffPolicy, DumbTestButYouGetTheIdea) {
  for (auto i = 0; i != 20; ++i) {
    ExponentialBackoffPolicy tested(ms(0), ms(1001), ms(1001), 1.001, 1.001);
    auto delay = tested.OnCompletion();
    if (delay < ms(1000)) GTEST_SUCCEED();
  }
  GTEST_FAIL();
}
```

With the current code, the test would fail every run.

If the ranges are correct, the test would fail every 1001^20 runs.

Reply:
Addressed the test cases and got rid of the scaling check.

I *think* this addresses the edge cases.